core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -922,9 +923,21 @@ private void subtestCallsAndShutdown(boolean shutdownNow, boolean shutdownNowAft
     // Killing the remaining real transport will terminate the channel
     transportListener.transportShutdown(Status.UNAVAILABLE);
     assertFalse(channel.isTerminated());
+    assertNotEquals(TERMINATED, channel.getState(false));
+    final AtomicBoolean resourcesReleased = new AtomicBoolean();
```

Comment:
This is the "real test" of the new behavior, demonstrating how to be notified when everything's closed.

The answer to "How can I be notified asynchronously when gRPC resources are closed and no more network activity will happen?"

```java
CompletableFuture<Void> shutdown = new CompletableFuture();
channel.shutdown();
channel.notifyWhenStateChanged(SHUTDOWN, () -> shutdown.complete(null));
```